depends/libhdfs3/src/client/CryptoCodec.cpp

```diff
@@ -119,33 +119,38 @@ namespace Hdfs {
 			return -1;
 		}
 
-		//calculate new IV when appending a existed file
+		// Calculate iv and counter in order to init cipher context with cipher method. Default value is 0.
+		resetStreamOffset(crypto_method, stream_offset);
+
+		LOG(DEBUG3, "CryptoCodec init success, length of the decrypted key is : %llu, crypto method is : %d", AlgorithmBlockSize, crypto_method);
+		is_init = true;
+		return 1;
+
+	}
+
+	int CryptoCodec::resetStreamOffset(CryptoMethod crypto_method, int64_t stream_offset) {
```

Comment:
The function prototype defines that it could return 1, 0, -1, but there is no value 0 return.